src/main/scala/org/zalando/spearheads/innkeeper/RouteDirectives.scala

```diff
@@ -33,6 +31,17 @@ trait RouteDirectives {
       }
     }
 
+  def findRouteOwnerTeam(id: Long, routesService: RoutesService, requestDescription: String)(implicit executionContext: ExecutionContext): Directive1[TeamName] =
+    Directive[Tuple1[TeamName]] { inner => ctx =>
+      {
+        routesService.getOwnerTeamForRoute(id).fast.transformWith {
+          case Success(ServiceResult.Success(teamName))               => inner(Tuple1(teamName))(ctx)
+          case Success(ServiceResult.Failure(ServiceResult.NotFound)) => reject(RouteNotFoundRejection(requestDescription))(ctx)
```

Comment:
I guess this should reject with a different reason, as this is not a route. Please consider creating a `TeamNotFoundRejection`